addons/stock/models/stock_move.py

```diff
@@ -1109,6 +1109,11 @@ def _action_confirm(self, merge=True, merge_into=False):
             moves = self._merge_moves(merge_into=merge_into)
         # call `_action_assign` on every confirmed move which location_id bypasses the reservation
         moves.filtered(lambda move: not move.picking_id.immediate_transfer and move._should_bypass_reservation() and move.state == 'confirmed')._action_assign()
+        for move in move_to_confirm.filtered(lambda move: move in moves):
```

Comment:
I would rather do
``
for move in moves:
``

and modify the upper line
``
moves = self
``
by
``
moves = self.filtered(lambda m: m.state != 'draft')
``